src/haven/plans/robot_sample.py

```diff
@@ -0,0 +1,92 @@
+import logging
+
+from bluesky import plan_stubs as bps
+
+from ..instrument.instrument_registry import registry
+from ..motor_position import rbv
+
+log = logging.getLogger(__name__)
+
+
+__all__ = ["robot_sample"]
+
+
+def robot_sample(robot, sampleN, *args):
+    """
+    Use robot to load sampleN at a fixed Aerotech stage position or any
+    motors.
+
+    e.g.
+    robot_sample(robot, 9, motor1, 100, motor2, 200, motor3, 50)
+
+    Parameters:
+    - robot: robot device
+    - sampleN: Sample number
+    - *args: multiple pairs of motor and pos
+    """
+
+    robot = registry.find(robot)
+
+    # Check if power is on
+    # if robot.power_rbv.get() == Off:
+    #    raise ValueError("Robot is NOT powered. Please turn on the power.")
+
+    # Check if remote control is on
+    # if not robot.remote_control.get() == Off:
+    #    raise ValueError(
+    #        "Robot is NOT in reomete control mode. Please click on Pad for remote control."
+    #    )
+
+    # Check if gripper is on
+    # if robot.gripper_activated.get() == Off:
+    #    raise ValueError("Gripper is not activated. Please activate the gripper.")
+
+    # Check if robot is running
+    # if robot.grogram_running.get() == On:
+    #   raise ValueError("Robot is running now.")
+
+    # Find motors
+    motor_list = [registry.find(motor) for motor in args[::2]]
+
+    # Record the motor positions before load sampls
+    initial_positions = [rbv(motor) for motor in motor_list]
+
+    # Move the aerotech to the loading position
+    for motor, pos in zip(args[::2], args[1::2]):
```

Comment:
Should this iterate over ``motor_list`` instead of ``args[::2]``? Otherwise, the motors might still be strings since they haven't been looked up in the registry.

Also, you might think about having ``new_positions = args[::2]`` up where you define ``motor_list`` on line 49. It's not immediately obvious what ``zip(args[::2], args[1::2])`` does but ``zip(motor_list, new_positions)`` is clear.